src/script/descriptor.cpp

```diff
@@ -713,6 +716,51 @@ std::unique_ptr<PubkeyProvider> ParsePubkeyInner(const Span<const char>& sp, boo
     if (extkey.key.IsValid()) {
         extpubkey = extkey.Neuter();
         out.keys.emplace(extpubkey.pubkey.GetID(), extkey.key);
+
+        if (!hardened && type != DeriveType::HARDENED && path.size() > 0) {
+            KeyOriginInfo last_hardened_info;
+            CKeyID extpubkeyid = extpubkey.pubkey.GetID();
+            std::copy(extpubkeyid.begin(), extpubkeyid.begin() + 4, last_hardened_info.fingerprint);
+
+            KeyPath::iterator it = path.end() - 1;
+            for (; it >= path.begin(); it--) {
```

Comment:
Travis doesn't like this: https://travis-ci.org/bitcoin/bitcoin/jobs/651243106#L4064

```
Error: attempt to decrement a dereferenceable (start-of-sequence) iterator.
```

Reply:
I was able to use `std::find_if` for this.